### DISCUSSION TITLE
Pulling consumers does not support DeliverGroup

### DISCUSSION THREAD

**Discussion by yang-xiaodong** [nats-io/nats.net#685] [Q&A]
I see that push-based consumers are not currently supported, and I was wondering if pull-based consumers support DeliverGroup for effects similar to Kafka Consumer Groups (A event sent to multiple groups, with load balancing of consumers within groups).

**Comment by mtmk** [comment]
thanks for bringing this up @yang-xiaodong we are missing pull consumer implementation we should implement them eventually. up to now we said we wouldn't have to expose the internals of consumer mechanics but this is a good example that we have to expose push consumers.

**Reply by mtmk** [reply]
see
* #686

**Comment by YashasAnand** [comment]
@mtmk , as @yang-xiaodong has mentioned regarding kafka's implementation of consumer groups, nats does support jetstream pull consumers with queue group, This is as designed I am assuming. 

If we have orders consumer, I can create 3 order consumers with same name & filter subject , and then any message to orders will be load-balanced internally by server.  Thus only 1 consumer ends up getting this message.

But what if I want this order message to be consumed by all 3 maybe to update local cache on 3 applications running 3 consumers on same filter subject. 

In kafka i can achive this by just keeping all 3 consumers in different consumer groups but subscribed on same topic.

Pls could you suggest what can be done here? We dont want to use different filter subject.

**Reply by mtmk** [reply]
You can load balance using a single consumer created on the server and have multiple consuming applications/clients which would receive different messages spread across them.

you can also create multiple consumers on the server to consume the messages separately by multiple consuming applications/clients using different consumer each on the server.

**Reply by YashasAnand** [reply]
Lets say i am using jetstream & i have a limit based retention policy, if i have one subject orders.pizza & i have 3 consumers running on 3 application servers which need to consume orders.pizza to do some business logic internally. Currntly with what ive tried, i have created 3 consumers with ConsumerConfig Filter subject as orders.pizza. 

I receive this only in 1 app. @mtmk Pls could you elobrate what you have explained above.
